src/Agda2Hs/Compile/Type.hs

```diff
@@ -104,8 +104,7 @@ compileTopLevelType t cont = do
           underAbstraction a atel $ \tel ->
             go tel (cont . constrainType c)
       | otherwise = underAbstraction a atel $ \tel -> do
-          unlessM (asks isCompilingInstance) $
```

Comment:
This was the only other place using `isCompilingInstance` not pertaining to copatterns. As it appears, it's not even related to instances, but rather to whether or not the generated Haskell type will be used in the output code.
When compiling a class instance, we discard the type and thus shouldn't enable the language extension. But that's not the only place where we discard the type (Every `compileFun False` does the same), so I added a flag. A bit ad-hoc though.